sections/Deals/AcceptOrReject/AcceptOrRejectBox/AcceptOrRejectBox.tsx

```diff
@@ -0,0 +1,298 @@
+import Image from 'next/image';
+import styled from 'styled-components';
+import { FC, useEffect, useState, useMemo } from 'react';
+
+import Info from 'assets/svg/info.svg';
+
+import Connector from 'containers/Connector';
+import TransactionData from 'containers/TransactionData';
+
+import { Status } from 'components/DealStatus';
+import QuestionMark from 'components/QuestionMark';
+import { FlexDivRowCentered, Tooltip } from 'components/common';
+import ConfirmTransactionModal from 'components/ConfirmTransactionModal';
+
+import {
+	Container,
+	ContentContainer,
+	ActionBoxInputLabel,
+	InputContainer,
+	ActionBoxInput,
+	ActionBoxMax,
+	ActionBoxHeaderWrapper,
+	ActionBoxHeader,
+	FlexDivCenterRow,
+	ActionButton,
+} from 'sections/shared/common';
+
+import { GasLimitEstimate } from 'constants/networks';
+import { statusToText, swimmingPoolID } from 'constants/pool';
+import { TransactionType } from 'constants/transactions';
+
+import AcceptOrRejectError from '../AcceptOrRejectError';
+
+interface AcceptOrRejectDealBoxProps {
+	poolId?: string;
+	onSubmit: () => void;
+	inputValue: number | string;
+	setInputValue: (val: number | string) => void;
+	setIsMaxValue: (val: boolean) => void;
+	userPoolBalance: string | null;
+	gasLimitEstimate: GasLimitEstimate;
+	purchaseCurrency: string | null;
+	dealRedemptionData: {
+		status: Status;
+		maxProRata: number;
+		isOpenEligible: boolean;
+		purchaseTokenTotalForDeal: number;
+		totalAmountAccepted: number;
+	};
+}
+
+const AcceptOrRejectDealBox: FC<AcceptOrRejectDealBoxProps> = ({
+	poolId,
+	onSubmit,
+	inputValue,
+	setInputValue,
+	setIsMaxValue,
+	userPoolBalance,
+	gasLimitEstimate,
+	purchaseCurrency,
+	dealRedemptionData,
+}: AcceptOrRejectDealBoxProps) => {
+	const { walletAddress } = Connector.useContainer();
+	const { setGasPrice, txType, setTxType } = TransactionData.useContainer();
```

Comment:
It's used below to handle what kind of tx will do

Reply:
It's not? It's literally an extra step we don't need anymore.

Reply:
it is used to handle that for Accept and Withdraw but since that is in a single component now and not touching all the transactions we can change that out for a simple boolean. it was more important when all the transactions were in a single action box which is no longer the case

Reply:
Done